pkg/plugins/manager/testdata/external-registration/plugin.json

```diff
@@ -18,16 +18,6 @@
     "version": "1.0.0"
   },
   "externalServiceRegistration": {
-    "impersonation": {
-      "enabled" : true,
-      "groups" : true,
-      "permissions" :  [
-        {
-          "action": "read",
-          "scope": "datasource"
-        }
-      ]
-    },
     "self": {
```

Comment:
does it make sense to change this to `serviceAccount` or similar? I don't know if the `self` key makes sense in the service account-only scenario. It's still behind the feature flag so we can make breaking changes if needed

Reply:
I agree `self` sounds weird in the service account-only scenario. However `serviceAccount` sounds a little strange to me in the OAuth2 Server scenario.
In favor of the `serviceAccount` key: we might remove the OAuth2 server code in favor of another approach.

Reply:
Can we challenge this even further.
Should we drop the self key altogether and move things up one layer:
```json
"externalServiceRegistration": {
  "permissions": {"action":  "datasources:read", "scope": "datasources:*"}
}
```

And in the case of OAuth2:
```json
"externalServiceRegistration": {
  "permissions": {"action":  "datasources:read", "scope": "datasources:*"},
  "impersonation": {
     "permissions": {"action": "folders:read", "scope": "folders:*"}
   }
}
```

Reply:
the simpler the better if we are not losing functionality. Last one sounds good to me :+1: